### PR TITLE
fix(NES-1636): translate journey when copying to another team

### PR DESCRIPTION
**Pull request by jianwei1**
## Summary

When a user copied a journey to another team and enabled the AI translation toggle, the journey was duplicated but never translated — it landed in the destination team in the source language, with no error or feedback. The dialog (`CopyToTeamDialog`) had been forwarding `language` and `showTranslation` to its `submitAction` since PR #6761, but the consumer in `DuplicateJourneyMenuItem` only consumed `teamId` and silently dropped the rest. TypeScript could not catch the structural mismatch.

After this fix, `DuplicateJourneyMenuItem` mirrors the working pattern from `TranslateJourneyDialog`: after `journeyDuplicate` resolves, `useJourneyAiTranslateSubscription` is started against the duplicated journey id and the user-selected target language. The subscription's `onComplete` closes the dialog, refetches template stats when applicable, and surfaces the success snackbar. The same-team direct-duplicate path (no dialog) and the translation-off path are unchanged.

Linear: [NES-1636](https://linear.app/jesus-film-project/issue/NES-1636/ai-translation-does-not-work-when-copying-a-journey-to-another-team) — bug in Production
Plan: [`docs/plans/2026-05-05-001-fix-nes-1636-ai-translate-on-copy-to-team-plan.md`](docs/plans/2026-05-05-001-fix-nes-1636-ai-translate-on-copy-to-team-plan.md)

## What changed

- `DuplicateJourneyMenuItem.handleDuplicateJourney` now accepts `(teamId?, language?, showTranslation?)` and conditionally fires the AI translate subscription after `journeyDuplicate` resolves.
- `useJourneyAiTranslateSubscription` is wired with `onComplete` (refetch template stats, success snackbar, close dialog, call `handleCloseMenu`) and `onError` (surface error snackbar with a `t('Failed to translate journey')` fallback when the upstream message is empty).
- `isTranslating` and `translationProgress` are passed back to `CopyToTeamDialog` so its existing in-dialog progress UI renders during translation.
- Source-language name derives from `journey?.language.name` on the prop, with `useJourney()` as fallback. The existing `GetAdminJourneys` query already returns `language.name` and `title`, so no GraphQL change.

## Test coverage

4 new tests added under a new `AI translation on copy to team` describe block, plus 1 strengthened error-path assertion. All 10 tests pass.

| Scenario | What it pins |
| --- | --- |
| Translate-on happy path | Duplicate runs, subscription starts with the duplicated id and selected target language, dialog closes after `onComplete`, success snackbar shows |
| Subscription error keeps the duplicate | Error snackbar shows; `handleCloseMenu` and `refetchTemplateStats` are NOT called; the duplicate is not rolled back |
| `fromTemplateId` refetch on translate complete | `refetchTemplateStats` is called with the template id after the subscription completes |
| `journeyDuplicate` error path | "Failed to duplicate journey" snackbar shows; subscription is NOT invoked; `handleCloseMenu` is NOT called |

The 5 existing tests (same-team direct-duplicate, dialog-shared-with-me path with and without journey prop, dialog cancel, `refetchTemplateStats` on direct duplicate) continue to pass unchanged, locking in the no-regression contract.

## Residual review findings

`ce-code-review` autofix applied 3 safe local fixes (empty-error-message fallback, error-path assertion strengthening, duplicate-error-path test) and surfaced 16 residual findings recorded in [`docs/residual-review-findings/jianweichong-nes-1636-fix-ai-translate-on-copy-to-team.md`](docs/residual-review-findings/jianweichong-nes-1636-fix-ai-translate-on-copy-to-team.md) for reviewer attention. Highlights:

- **#1 (P1)** Race: `GetAdminJourneys` refetch from `updateLastActiveTeamId.onCompleted` can unmount the menu item before the subscription's `onComplete` fires, leaving an orphan untranslated journey with no UI feedback. The same hazard exists in `TranslateJourneyDialog` — shared subscription-lifecycle debt.
- **#2 (P1)** Cancel button mid-translation: dialog `onClose` only sets `open=false` but leaves `translationVariables` set, so the subscription keeps running and `onComplete` fires snackbar/`handleCloseMenu` on a dismissed dialog.
- 14 P2/P3 items covering: missing client-side timeout, error-leaves-orphan-on-resubmit, snackbar wording inconsistency, `journeyDuplicate: null` partial response handling, `TranslationVariables` type duplication, and a state-machine refactor that would consolidate `setLoading(false)` calls scattered across 5 paths.

None of these block the user-visible bug fix; they document follow-up work the reviewer can weigh against scope.

---

[Compound Engineering](https://github.com/EveryInc/compound-engineering-plugin)
Claude Code

## Summary by CodeRabbit

* **New Features**
  * Duplicate-to-team flow can optionally translate journeys during copy and shows translation progress/status in the copy dialog.

* **Bug Fixes**
  * Ensures translation starts and completes correctly after duplication, surfaces success/error notifications, avoids starting translation on duplicate failure or missing IDs, and refetches template stats after translated copies.

* **Tests**
  * Added/expanded tests covering translation-on-copy success, failures, progress, and edge cases; updated test imports.

* **Documentation**
  * Added plan and residual-review notes detailing design, risks, and test scenarios.

### CODE REVIEW

File: apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/DuplicateJourneyMenuItem/DuplicateJourneyMenuItem.tsx [hunk 1]
```diff
@@ -115,10 +206,13 @@ export function DuplicateJourneyMenuItem({
       <CopyToTeamDialog
         title={t('Copy to Another Team')}
         open={open}
+        loading={loading}
         onClose={() => {
           setOpen(false)
         }}
         submitAction={handleDuplicateJourney}
+        isTranslating={translationVariables != null}
+        translationProgress={translationProgress}
```
[the remark this answers is not in the document]
Comment (jianwei1) [reply apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/DuplicateJourneyMenuItem/DuplicateJourneyMenuItem.tsx:228]: Skipping this one — the suggested fix is for a code path that isn't user-reachable.

`TranslationDialogWrapper` conditionally hides both the Cancel and Create buttons while `loading` is true:

```tsx
dialogActionChildren={
  <>
    {!loading && (
      <>
        <Button ... onClick={onClose}>Cancel</Button>
        <Button ... onClick={onTranslate}>Create</Button>
      </>
    )}
  </>
}
```

During translation we keep `loading` true (intentionally, until `onComplete` / `onError` fires), so neither button is rendered — the body is replaced by the progress bar. `CopyToTeamDialog.handleDialogClose` also blocks backdrop click and Escape while `loading || isTranslating`. The only paths that fire `onClose` while `translationVariables` is set are programmatic ones that don't exist in production today.

Adding the cleanup in `onClose` would be dead code and the comment would imply a cancel-mid-translation flow that the UI doesn't support. Out of scope for NES-1636 (which is purely about wiring the translation step that was previously dropped). If we ever expose a Cancel / Stop button during translation, the cleanup belongs alongside that change so the wiring matches the user-visible affordance.

File: docs/plans/2026-05-05-001-fix-nes-1636-ai-translate-on-copy-to-team-plan.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+## Open Questions
+
+### Resolved During Planning
+
+- **Does `GetAdminJourneys` expose journey language name?** Yes — confirmed at `apps/journeys-admin/src/libs/useAdminJourneysQuery/useAdminJourneysQuery.ts:8-50`. No GraphQL change required.
+- **Should the same-team direct duplicate (no dialog) ever translate?** No — the in-team auto-duplicate path (lines 105-112) bypasses the dialog entirely; the user never sees a translation toggle, so showTranslation is implicitly false. Maintain the existing zero-arg call shape.
+
+### Deferred to Implementation
+
+- **Whether to also pass translation context up to `handleCloseMenu` timing.** The current `handleDuplicateJourney` calls `handleCloseMenu()` as soon as the duplicate succeeds (line 69). For the translate path, closing the menu wrapper before the subscription completes is fine (the dialog stays open via `TranslationDialogWrapper`'s loading state), but verify in implementation that closing the menu doesn't unmount the dialog and abort the subscription. If it does, defer `handleCloseMenu` until the subscription `onComplete` fires.
```
[the remark this answers is not in the document]
Comment (jianwei1) [reply docs/plans/2026-05-05-001-fix-nes-1636-ai-translate-on-copy-to-team-plan.md:86]: Skipping this — the plan doc is a planning snapshot, and the implementation already chose Option A (defer `handleCloseMenu` to the subscription's `onComplete`).

Look at the merged file:

```tsx
const { data: translationData } = useJourneyAiTranslateSubscription({
  variables: translationVariables,
  skip: translationVariables == null,
  onError(error) { ... setLoading(false); setTranslationVariables(undefined) },
  onComplete() {
    if (fromTemplateId != null) void refetchTemplateStats([fromTemplateId])
    enqueueSnackbar(...)
    setTranslationVariables(undefined)
    setLoading(false)
    setOpen(false)
    handleCloseMenu()       // <- only fires after the subscription resolves
  }
})
```

The translation branch of `handleDuplicateJourney` returns early after `setTranslationVariables(...)` without calling `handleCloseMenu` itself. The non-translation branch calls `handleCloseMenu` synchronously, which is safe because no subscription is in flight there. The deferred-question paragraph in the plan was about validating this choice during implementation, and it landed correctly. Updating the plan retroactively isn't worth the churn — the code is now the authoritative artifact.

File: apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/DuplicateJourneyMenuItem/DuplicateJourneyMenuItem.tsx [hunk 3]
```diff
@@ -30,29 +48,105 @@ export function DuplicateJourneyMenuItem({
...
+
+        setTranslationVariables({
+          journeyId: duplicatedJourneyId,
+          name: journeyData.title ?? '',
+          journeyLanguageName: sourceLanguageName,
+          textLanguageId: language.id,
+          textLanguageName: language.nativeName ?? language.localName ?? '',
+          userLanguageId: journeyData.language.id,
+          userLanguageName: sourceLanguageName
         })
```
[the remark this answers is not in the document]
Comment (jianwei1) [reply apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/DuplicateJourneyMenuItem/DuplicateJourneyMenuItem.tsx:146]: Skipping for scope. The `sourceLanguageName` derivation is a deliberate mirror of `TranslateJourneyDialog.tsx:158-159`:

```tsx
const currentLanguageName =
  journeyData.language.name.find(({ primary }) => !primary)?.value ?? ''
```

That pattern has been in production since the AI translation feature shipped — same fall-through-to-empty for journeys whose `language.name` only carries a primary entry. NES-1636 was deliberately framed as a near-mechanical port of `TranslateJourneyDialog`'s wiring (per the plan's "Mirror TranslateJourneyDialog's subscription wiring rather than introducing a shared hook" decision), so fixing this here only would diverge from the reference pattern: same input would now produce different translation prompts depending on which entry point fired.

The right shape for this fix is a follow-up that updates **both** callsites and their tests in lockstep — happy to file it as a separate ticket if translation quality from missing source-language context is showing up in QA. For this PR I'd rather hold the symmetry.
